tests/e2e/specs/wcpay/shopper/shopper-bnpls-checkout.spec.js

```diff
@@ -49,21 +52,16 @@ describe.each( cardTestingPreventionStates )(
 
 		describe.each( bnplProviders )(
 			`Checkout with %s, carding protection ${ cardTestingPreventionEnabled }`,
-			( providerName ) => {
-				it( `should successfully place order with ${ providerName }`, async () => {
+			( providerId ) => {
+				it( `should successfully place order with ${ providerId }`, async () => {
 					await shopperWCP.emptyCart();
 					await setupProductCheckout(
 						config.get( 'addresses.customer.billing' ),
 						[ [ 'Beanie', 3 ] ]
 					);
 					await uiUnblocked();
 					// Select BNPL provider as payment method.
-					const xPathPaymentMethodSelector = `//*[@id='payment']/ul/li/label[contains(text(), '${ providerName }')]`;
-					await page.waitForXPath( xPathPaymentMethodSelector );
-					const [ paymentMethodLabel ] = await page.$x(
-						xPathPaymentMethodSelector
-					);
-					await paymentMethodLabel.click();
+					await selectOnCheckout( providerId, page );
```

Comment:
Thanks for refactoring this, I know this suite in particular is flaky so hopefully this improves the reliability.